h1st/h1st/core/ensemble.py

```diff
@@ -2,106 +2,71 @@
-
-        :param prepared_data: Dictionary object contains train data, labels and validation data.
-        """
-        train_data = prepared_data['train_data']
-        train_labels = prepared_data['train_labels']
-        all_inputs = None
-        all_labels = None
+        # Feed input_data to each sub-model and get predictions 
+        predictions = [m.predict({'X': X})['predictions'] for m in self._sub_models]
         
-        for each_input, each_label in zip(train_data, train_labels):
-            inputs = None
-            labels = each_label
-            for model in self.models:
-                df_prediction = self._extract_prediction(model.predict(each_input))
-                
-                df_prediction = df_prediction[self.prediction_columns]
-                df_prediction.fillna(0, inplace=True)
-                p1 = df_prediction.values
-                if inputs is None:
-                    inputs = p1
-                else:
-                    inputs = np.hstack((inputs, p1))
-            if all_inputs is None:
-                all_inputs = inputs
-            else:
-                all_inputs = np.vstack((all_inputs, inputs))
-            if all_labels is None:
-                all_labels = labels
-            else:
-                all_labels = np.vstack((all_labels, labels))
-
-        # assumption: all_inputs ~ numpy array, all_labels ~ numpy array
-        self.model.fit(all_inputs, all_labels)
-
-
-class MultiOutputClassifierEnsemble(StackEnsemble):
-    def __init__(self, models):
+        print(X.shape, predictions[0].shape, predictions[1].shape)
+        # Combine raw_data and predictions
+        X = np.hstack([X] + predictions)
+
+        # Scale data with RobustScaler
+        if training:
+            self.stats = RobustScaler(quantile_range=(5.0, 95.0), with_centering=False).fit(X)
+        X = self.stats.transform(X)
+        return X
+
+    # def prep_data(self, data: dict):
+    #     # Preprocess raw input_data
+    #     X_train = data['X_train']
+    #     y_train = data['y_train']
+    #     return {'X_train': X_train, 'y_train': y_train}
+
+    def train(self, prepared_data: dict):
+        X_train, y_train = (prepared_data['X_train'], prepared_data['y_train'])
+        X_train = self.preprocess(X_train, training=True)
+        self.ensembler.fit(X_train, y_train)
+
+    def predict(self, data: dict):
+        X = data['X']
+        X = self.preprocess(X)
+        return {'predictions': self.ensembler.predict(X)}
+
+
+class StackEnsembleClassifier(StackEnsemble):
+    def evaluate(self, data: dict, metrics=['accuracy']):
+        X_test = data['X_test']
+        y_test = data['y_test']
+        y_pred = self.predict({'X': X_test})['predictions']
+        # print(y_test.shape, y_pred.shape)
+        self.metrics['confusion_matrix'] = confusion_matrix(y_test, y_pred)
+        self.metrics['recall'], self.metrics['recall'], self.metrics['f1'] , self.metrics['support'] \
```

Comment:
I am not sure whether this is what you want it to be or a typo issue.
self.metrics['recall'], self.metrics['recall']

If it is a type issue, it should be 
self.metrics['precision'], self.metrics['recall']